src/lib/optimizer/join_ordering/join_graph.cpp

```diff
@@ -73,8 +79,13 @@ std::vector<std::shared_ptr<AbstractExpression>> JoinGraph::find_join_predicates
   std::vector<std::shared_ptr<AbstractExpression>> predicates;
 
   for (const auto& edge : edges) {
-    if ((edge.vertex_set & vertex_set_a).none() || (edge.vertex_set & vertex_set_b).none()) continue;
-    if (!edge.vertex_set.is_subset_of(vertex_set_a | vertex_set_b)) continue;
+    if ((edge.vertex_set & vertex_set_a).none() || (edge.vertex_set & vertex_set_b).none()) {
+      continue;
+    }
+
+    if (!edge.vertex_set.is_subset_of(vertex_set_a | vertex_set_b)) {
+      continue;
+    }
```

Comment:
not sure if we want to combine both cases in one if, they are already quite long